<?xml version="1.0" encoding="utf-8"?>
<robot name="sim_pusher" xmlns:xacro="http://wiki.ros.org/xacro">
  <material name="red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>

  <link name="base_link">
	 <inertial>
     <mass value="1.0"/>
     <inertia ixx="1.0" ixy="0" ixz="0" iyy="1.0" iyz="0" izz="1.0"/>
     <origin rpy="0 0 0" xyz="0 0 0"/>
    </inertial>
  </link>

	<joint name="pusher_joint_x" type="prismatic">
    <parent link="base_link"/>
    <child link="dummy_link"/>
    <axis xyz="1 0 0"/>
    <origin rpy="0 0 0" xyz="0 0 0.06"/>
    <limit effort="0" velocity="1000.0"/>
  </joint>

	<link name="dummy_link">
	 <inertial>
     <mass value="1.0"/>
     <inertia ixx="1.0" ixy="0" ixz="0" iyy="1.0" iyz="0" izz="1.0"/>
     <origin rpy="0 0 0" xyz="0 0 0"/>
    </inertial>
  </link>

	<joint name="pusher_joint_y" type="prismatic">
    <parent link="dummy_link"/>
    <child link="pusher_link"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <limit effort="0" velocity="1000.0"/>
  </joint>

  <link name="pusher_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="1"  ixy="0"  ixz="0" iyy="1" iyz="0" izz="1" />
    </inertial>

    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.05"/>
      </geometry>
      <material name="red"/>
    </visual>

    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.05"/>
      </geometry>
    </collision>
  </link>

</robot>
